public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { FormEvent, useState } from 'react';
+import { Button, Icon, Input, Label, RadioButtonGroup, useStyles } from '@grafana/ui';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import { css, cx } from '@emotion/css';
+import { debounce } from 'lodash';
+
+import { PromAlertingRuleState } from 'app/types/unified-alerting-dto';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import DataSourceSelect from '../DataSourceSelect';
+import { getFiltersFromUrlParams } from '../../utils/misc';
+
+const RulesFilter = () => {
+  const [queryParams, setQueryParams] = useQueryParams();
+  // This key is used to force a rerender on the inputs when the filters are cleared
+  const [filterKey, setFilterKey] = useState<number>(Math.floor(Math.random() * 100));
+  const dataSourceKey = `dataSource-${filterKey}`;
+  const queryStringKey = `queryString-${filterKey}`;
+
+  const { dataSource, alertState, queryString } = getFiltersFromUrlParams(queryParams);
+
+  const styles = useStyles(getStyles);
+  const stateOptions = Object.entries(PromAlertingRuleState).map(([key, value]) => ({ label: key, value }));
+
+  const handleDataSourceChange = (dataSourceValue: SelectableValue<string>) => {
+    setQueryParams({ dataSource: dataSourceValue.value });
+  };
+
+  const handleQueryStringChange = debounce((e: FormEvent<HTMLInputElement>) => {
+    const target = e.target as HTMLInputElement;
+    setQueryParams({ queryString: target.value || null });
+  }, 600);
+
+  const handleAlertStateChange = (value: string) => {
+    setQueryParams({ alertState: value });
```

Comment:
clicking the same state button  twice should un-check it; now if you filter by state, the only way to undo it and see all rules is to use "clear filters"

Reply:
We'll likely have to use a different component or create our own for this instance. The `RadioButtonGroup` only allows for one value to be selected at a time and the onChange handler only fires when selecting a different value.

Reply:
Right. Would probably need to invent a `CheckboxButtonGroup` then... Let's leave this for later